internal/httpclient/auth_oauth2.go

```diff
@@ -38,6 +38,25 @@ func (oauth oauth2Config) Client(ctx context.Context, base *http.Client) *http.C
 		return base
 	}
 
+	// Support for refresh_token grant type with bootstrapped refresh token to obtain access token
+	if gt, ok := oauth.EndpointParams["grant_type"]; ok && gt[0] == "refresh_token" {
+		conf := &oauth2.Config{
+			ClientID:     oauth.ClientKey,
+			ClientSecret: oauth.ClientSecret,
+			Endpoint: oauth2.Endpoint{
+				TokenURL:  oauth.TokenURL,
+				AuthStyle: oauth2.AuthStyleAutoDetect,
+			},
+			Scopes: oauth.Scopes,
+		}
+		// We don't consider bootstrapped access token if any as it might be expired, rather we generate a new one
+		token := &oauth2.Token{}
+		if rt, ok := oauth.EndpointParams["refresh_token"]; ok {
+			token.RefreshToken = rt[0]
+		}
+		return conf.Client(ctx, token)
```

Comment:
```suggestion
		return conf.Client(context.WithValue(ctx, oauth2.HTTPClient, base), token)
```

I haven't tried this ^ but otherwise I'm not sure how we'd carry the base configuration through.

Reply:
If we don't provide one, it uses the default one. So, it wasn't caught in our testing. Anyway, I'll update this if a custom client is provided; we need to use the same. Thanks!

Reply:
Please review @Jeffail